addons/mail/static/src/components/message/message_tests.js

```diff
@@ -197,17 +197,21 @@ QUnit.test('Notification Sent', async function (assert) {
             model: 'mail.channel',
         }]],
     });
-    const message = this.env.models['mail.message'].create({
+    const messageData = {
         id: 10,
         message_type: 'email',
-        notifications: [['insert', {
+        model: 'mail.channel',
+        notifications: [{
             id: 11,
             notification_status: 'sent',
             notification_type: 'email',
-            partner: [['insert', { id: 12, name: "Someone" }]],
-        }]],
-        originThread: [['link', threadViewer.thread]],
-    });
+            res_partner_id: [ 12, "Someone" ],
+        }],
+        res_id: 11,
+    };
+    const message = this.env.models['mail.message'].create(
+        this.env.models['mail.message'].convertData(messageData)
+    );
```

Comment:
Should define the message in `this.data` and then call/await `performRpcMessageFetch` in the test to get the message

Reply:
also need the channel in `this.data` for the rpc to work.

Reply:
Alternative is that threadCache (through thread viewer) will also call this RPC already, but it is harded to `await` on it because it happens inside a computed field.

If we had model event/obeserver we could just wait for `isLoaded` of `threadViewer.threadCache` to become true

Reply:
Ideally most tests should define stuffs in `data`, not manually create JS records. This should be the way to go for new tests at least.

I don't like explicit use of `convertData` in test. The fix in `convertData` should be reflected in processing of message data in some flow, therefore having `mail.message` mocked server data should make sure flow using `convertData` is good.

Reply:
Yep that's what I tried but this does not work as data are not fetched on component creation so as discussed with Seb we need to create a thread viewer component or to call thread cache method to fetch messages AND be sure to wait that they are loaded before starting the tests

Reply:
Ok, I see...

Last week, @seb-odoo had somewhat similar issues and suggested the idea of using Component's `trigger` to await properly. We are considering replacing `afterNextRender` with something more robust like these Component's `trigger`. I'm asking myself whether we shouldn't have a logging system at models level:

- full integration with models
- generalization of component hints
- can be used in tests (proper awaits)
- could help a lot for debugging (history of dynamic changes)

This is still being reflected...

Honestly, I prefer keeping most tests clean, so here I prefer setup with actual fields than message_format data + convert. If you really want to have a test appropriate for fix right now, I'm good with having a technical version of this test that uses `convertData`.